language/move-prover/bytecode/src/read_write_set_analysis.rs

```diff
@@ -0,0 +1,645 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    access_path::{AbsAddr, AccessPath, Addr, FootprintDomain, Offset, Root},
+    access_path_trie::AccessPathTrie,
+    compositional_analysis::{CompositionalAnalysis, SummaryCache},
+    dataflow_analysis::{AbstractDomain, DataflowAnalysis, JoinResult, TransferFunctions},
+    function_target::{FunctionData, FunctionTarget},
+    function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder, FunctionVariant},
+    stackless_bytecode::{Bytecode, Constant, Operation},
+};
+use move_model::{
+    ast::TempIndex,
+    model::{FunctionEnv, GlobalEnv, ModuleId, StructId},
+    ty::Type,
+};
+use std::{cmp::Ordering, fmt, fmt::Formatter};
+use vm::file_format::CodeOffset;
+
+/// The read/write set analysis is a compositional analysis that starts from the leaves of the
```

Comment:
nit: I think you need to use `//!` for module level documentation comments. Technically, if not, this text is associated with the next item which is `enum Access`. My understanding is `//!` allows to associate doc with the enclosing item (the module).